### ISSUE TITLE
System error 67 when attempting to mount

### ISSUE THREAD

**Issue by smares** [billziss-gh/sshfs-win#155]
I am using the latest versions as of today under Windows 10 Enterprise build 2004 and whenever I try to mount a folder, I get system error 67, the network name was not found (loose translation from German). I can ping the host fine however and can also connect to it via SSH without problem using PuTTY.

Tried with `net use Z: \\sshfs.k\www-data@container-host`

Also tried without the `.k` by using a password an no luck. I can connect via PuTTY to `container-host` using the `www-data` user and either a password or keyfile fine.

**Comment by smares** [comment]
To clarify, I am using `winfsp-1.6.20027.msi` and `sshfs-win-3.5.20024-x64.msi`

**Comment by billziss-gh** [comment]
This is indeed weird. Can you post the output of the file `diag.bat` that can be found under `C:\Program Files (x86)\WinFsp\bin`?

**Comment by smares** [comment]
```
WINFSP INSTALLATION DIRECTORY AND LAUNCHER REGISTRATIONS

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp
    InstallDir    REG_SZ    C:\Program Files (x86)\WinFsp\

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x1

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.k
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x0

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.r
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x1


WINFSP DLL REGISTRATIONS

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order
    ProviderOrder    REG_SZ    WinFsp.Np,RDPNP,LanmanWorkstation,webclient


HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\WinFsp.Np\NetworkProvider
    Name    REG_SZ    Windows File System Proxy
    ProviderPath    REG_SZ    C:\Program Files (x86)\WinFsp\bin\winfsp-x64.dll
    DeviceName    REG_SZ    \Device\WinFsp.Mup


HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\Application\WinFsp
    EventMessageFile    REG_SZ    C:\Program Files (x86)\WinFsp\bin\winfsp-x64.dll
    TypesSupported    REG_DWORD    0x7


WINFSP FSD CONFIGURATION AND STATUS

SERVICE_NAME: WinFsp
        TYPE               : 2  FILE_SYSTEM_DRIVER
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 1077  (0x435)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
[SC] QueryServiceConfig ERFOLG

SERVICE_NAME: WinFsp
        TYPE               : 2  FILE_SYSTEM_DRIVER
        START_TYPE         : 3   DEMAND_START
        ERROR_CONTROL      : 1   NORMAL
        BINARY_PATH_NAME   : \??\C:\Program Files (x86)\WinFsp\bin\winfsp-x64.sys
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : WinFsp
        DEPENDENCIES       :
        SERVICE_START_NAME :

D:(A;;LCRP;;;WD)(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)

WINFSP LAUNCHER SERVICE CONFIGURATION AND STATUS

SERVICE_NAME: WinFsp.Launcher
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, ACCEPTS_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
[SC] QueryServiceConfig ERFOLG

SERVICE_NAME: WinFsp.Launcher
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 0   IGNORE
        BINARY_PATH_NAME   : "C:\Program Files (x86)\WinFsp\bin\launcher-x64.exe"
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : WinFsp.Launcher
        DEPENDENCIES       :
        SERVICE_START_NAME : LocalSystem

D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)

OS INFORMATION

Hostname:                                      KALO375A
Betriebssystemname:                            Microsoft Windows 10 Enterprise
Betriebssystemversion:                         10.0.19041 Nicht zutreffend Build 19041
Betriebssystemhersteller:                      Microsoft Corporation
Betriebssystemkonfiguration:                   Mitglied der Domäne/Arbeitsgruppe
Betriebssystem-Buildtyp:                       Multiprocessor Free
Registrierter Benutzer:                        FLOW7 GmbH
Registrierte Organisation:
Produkt-ID:                                    00329-10458-00000-AA237
Ursprüngliches Installationsdatum:             27.02.2020, 17:39:42
Systemstartzeit:                               04.03.2020, 00:00:36
Systemhersteller:                              Hewlett-Packard
Systemmodell:                                  HP Z420 Workstation
Systemtyp:                                     x64-based PC
Prozessor(en):                                 1 Prozessor(en) installiert.
                                               [01]: Intel64 Family 6 Model 45 Stepping 6 GenuineIntel ~2701 MHz
BIOS-Version:                                  Hewlett-Packard J61 v03.94, 10.07.2018
Windows-Verzeichnis:                           C:\WINDOWS
System-Verzeichnis:                            C:\WINDOWS\system32
Startgerät:                                    \Device\HarddiskVolume1
Systemgebietsschema:                           de;Deutsch (Deutschland)
Eingabegebietsschema:                          de;Deutsch (Deutschland)
Zeitzone:                                      (UTC+01:00) Brüssel, Kopenhagen, Madrid, Paris
Gesamter physischer Speicher:                  65.460 MB
Verfügbarer physischer Speicher:               56.392 MB
Virtueller Arbeitsspeicher: Maximale Größe:    75.188 MB
Virtueller Arbeitsspeicher: Verfügbar:         65.747 MB
Virtueller Arbeitsspeicher: Zurzeit verwendet: 9.441 MB
Auslagerungsdateipfad(e):                      C:\pagefile.sys
Domäne:                                        Dom.Rastatt.de
Anmeldeserver:                                 \\KADC05
Hotfix(es):                                    3 Hotfix(e) installiert.
                                               [01]: KB4534170
                                               [02]: KB4537759
                                               [03]: KB4539080
Netzwerkkarte(n):                              1 Netzwerkadapter installiert.
                                               [01]: Intel(R) 82579LM Gigabit Network Connection
                                                     Verbindungsname: Ethernet
                                                     DHCP aktiviert:  Nein
                                                     IP-Adresse(n)
                                                     [01]: 10.45.60.95
                                                     [02]: fe80::b1c3:1177:428c:99fc
Anforderungen für Hyper-V:                     Erweiterungen für den VM-Überwachungsmodus: Ja
                                               Virtualisierung in Firmware aktiviert: Ja
                                               Adressübersetzung der zweiten Ebene: Ja
                                               Datenausführungsverhinderung verfügbar: Ja
```

**Comment by smares** [comment]
I am one step forward, by using `sshfs-win svc \sshfs.k\www-data@container-host Z:` manually, I am able to connect. My problem now is that my profile is on a network share, so is the `.ssh` directory with the keyfile and I am getting a warning that the permissions are too loose being 775. I have no idea how the Windows ACLs are mapped to UNIX rights.

**Comment by smares** [comment]
I am another step forward, I was able to delete all ACLs, including the inherited ones, for the keyfile, added my account as only user who is allowed to read the file and now `sshfs-win svc \sshfs.k\www-data@container-host Z:` works fine. However, I still cannot connect using `net use` or via the Windows Explorer Map network drive.

**Comment by smares** [comment]
![image](https://user-images.githubusercontent.com/818380/75893519-37c19d00-5e33-11ea-8198-581c41cf49ff.png)

**Comment by billziss-gh** [comment]
There was no obvious problem in the diagnostic output that you sent earlier.

Is the the account that you using when doing `net use` a local account or a domain account. Also if it is a local account is it part of the admin group on the local machine?

**Comment by smares** [comment]
It's a domain account but it is part of the local admin group.

**Comment by billziss-gh** [comment]
I suspect some kind of access issue between the domain account and the local SYSTEM account that will ultimately launch your file system.

Here is one thing to try (untested as I am currently away from a Windows machine):

```
cd "\Program Files (x86)\WinFsp\bin
launchctl-x64 start \sshfs.k\www-data@container-host Z:
```

Let me know what `launchctl` reports.

**Comment by smares** [comment]
KO launcher: error 161

Tried with both a regular cmd and a cmd opened with admin rights.

**Comment by billziss-gh** [comment]
Thanks.

Error 161 is `ERROR_BAD_PATHNAME`, which corresponds to the NT status code `STATUS_OBJECT_PATH_INVALID`.

Unfortunately it is not immediately obvious how this error/status code would be produced. WinFsp does not return this status code as far as I can remember (confirmed with a code search).

In any case this now looks less like an access control problem and more like a path validation problem. According to [this document](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file) the characters that you use in the path are all valid.

**Comment by smares** [comment]
I'll try to create an entry in my hosts file tomorrow with a short name without any special characters and see if it helps.

**Comment by smares** [comment]
Added an entry to my hosts file for "dev" and still no luck. :(

**Comment by facboy** [comment]
i'm getting the same error(s) with the latest versions.

**Comment by SergioLuis** [comment]
I'm also getting this very same error, in case I can help to diagnose the cause :)

**Comment by lehitoskin** [comment]
I get both System Error 67 as well as the KO launcher error 161. Running `sshfs-win.exe \user@server!port\folder Z:` directly will hang forever and I have to exit the process.

**Comment by billziss-gh** [comment]
@lehitoskin when you run `sshfs-win svc` from the command line, it waits for you to enter your password to log into the ssh server.

**Comment by lehitoskin** [comment]
@billziss-gh That's odd because it never prompts me for a username and password --- which it shouldn't ask for in the first place because I use a key to access the server.

**Comment by billziss-gh** [comment]
@lehitoskin 

If you use something like the following it should not ask for password:

```
sshfs-win svc \sshfs.k\USER@SERVER!PORT X:
```

But if you use something like the following it will ask for password, without prompting you. The reason is that you are using an internal command that is normally only supposed to be used by computers and not people.

```
sshfs-win svc \sshfs\USER@SERVER!PORT X:
```

(Please note that I did not test any of the above commands.)

**Comment by lehitoskin** [comment]
@billziss-gh Running `sshfs-win svc \sshfs.k\...` will just hang forever.

**Comment by billziss-gh** [comment]
@lehitoskin 

Please verify that the registry key `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\WinFsp\Services\sshfs.k` contains the value `Credentials=0`.

**Comment by lehitoskin** [comment]
@billziss-gh There is no key `sshfs.k` in that location.

**Comment by billziss-gh** [comment]
@lehitoskin are you using the 32-bit version of SSHFS-Win? In that case you should try the following key:

```
HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.
```

**Comment by lehitoskin** [comment]
I'm using the 64-bit version of SSHFS-Win and I do not have a WinFsp folder in that location.

**Comment by billziss-gh** [comment]
That can mean that you have uninstalled either WinFsp or SSHFS-Win. It could also mean that you have modified the registry by hand or that it was modified in some other way.

**Comment by lehitoskin** [comment]
Ah, call me a fool. I reinstalled both WinFsp and SSHFS-Win and now it works like a charm. Go figure...

**Comment by mvrk80** [comment]
I can confirm that the last windows 10 update (v 2004)  breaks the SSHFS-Win installations.
To fix that a fresh install of both SSHFS-Win and WinFsp is required

**Comment by billziss-gh** [comment]
@mvrk80 this is curious. When I updated to 2004 myself I did not experience this issue.

**Comment by mspielberg** [comment]
I am also experiencing problems mounting with an error 67, but the underlying cause appears to be different:

```
C:\Users\Miles>ver

Microsoft Windows [Version 10.0.18362.1016]

C:\Users\Miles>net use x: \\sshfs.k\orion@pathfinder
System error 67 has occurred.

The network name cannot be found.


C:\Users\Miles>"\Program Files (x86)\WinFsp\bin\launchctl-x64.exe" start \sshfs.k\orion@pathfinder.orionnet.net x:
KO CallNamedPipe = 2

C:\Users\Miles>"\Program Files\SSHFS-Win\bin\sshfs-win.exe" svc \sshfs.k\orion@pathfinder.orionnet.net x:
Warning: Permanently added 'pathfinder.orionnet.net,192.168.10.1' (ECDSA) to the list of known hosts.
The service sshfs has been started.
```

**Comment by simonvetter** [comment]
Just came here to report the same symptoms as @mspielberg:
- mounting through either net use, launchctl-64.exe or through the explorer UI all fail with "System error 67 has occurred" "The network name cannot be found",
- invoking sshfs-win.exe does work (authenticates, adds the X: mount point to the system and allows browsing and transferring files).

I am using ssh with pubkey auth, with the following command:
```
"C:\Program Files\SSHFS-Win\bin\sshfs-win.exe" svc \sshfs.k\user@host.fqdn x:
```
using sshfs-win v3.5.20160-x64 and winfsp v1.8.20304, installed from their msi packages.

The SSH client seems to be invoked as:
- I do see SSH traffic going on,
- /var/log/auth.log shows the client connecting and dropping the connection before authenticating ("Connection closed by authenticating user <user> <ip address> port 49748 [preauth]".

I haven't been able to find much information in the windows event log.

Given the pre-auth disconnect message, my hunch is that the ssh.exe process spawn through launchctl-64.exe may not have access to the private key file, but I can't be sure.

**Comment by bkarlson** [comment]
Exactly the same happening: `error 67` + sshfs-win.exe hanging with `The service sshfs has been started.` 

Fresh Win10 install + latest versions of sshfs-win and WinFsp. I made sure that ssh with pubkey works (took nearly an hour to figure out some obscure problem with `administrators_authorized_keys` permissions), yet I cannot mount the dir nor get any meaningful error message why it cannot happen.

p.s. I've been away from Windows for about 15 years, and I am sincerely blown away how buggy and unnecessary convoluted the system has become comparing to anything else on the market. To accomplish such a simple task (mount a dir over ssh to a number of servers) I have already spent 1 work day with not end of problems in sight, just an endless stream of unresolved issues on Github and unanswered questions on Q&A portals. Unfortunatelly, our current project requires Windows presence, but I cannot believe how messed up the whole thing is and cannot wait to get over with it.

/rant

**Comment by bkarlson** [comment]
While trying to diagnose the problem I can see nothing wrong in the log (see below), and `scp` works w/o any issues. 

```
 get_user_token - i am running as Administrator, returning process token
 user Administrator matched group list administrators at line 84
 match found
 reprocess config:85 setting AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys
 ssh_msg_recv entering
 monitor_apply_keystate: packet_set_state
 set_newkeys: mode 0
 rekey after 134217728 blocks
 set_newkeys: mode 1
 rekey after 134217728 blocks
 ssh_packet_set_postauth: called
 ssh_packet_set_state: done
 ssh_msg_recv entering
 notify_hostkeys: key 0: ssh-rsa SHA256:fzcPfYmpFgrA94YXfULN/jbz6sQ4K2cBEZZdPRhules
 notify_hostkeys: key 1: ecdsa-sha2-nistp256 SHA256:xj3ogUDX4NC1jsbXOd8nVVvnS7XXVsbreZ3H6MBxyNk
 notify_hostkeys: key 2: ssh-ed25519 SHA256:g9neRX1xfgnC45CGwQvI+DnH/a3bff1j5mVD6Pj2T1w
 notify_hostkeys: sent 3 hostkeys
 send packet: type 80
 active: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
 Entering interactive session for SSH2.
 fd 7 setting O_NONBLOCK
 fd 8 setting O_NONBLOCK
 server_init_dispatch
 receive packet: type 90
 server_input_channel_open: ctype session rchan 0 win 2097152 max 32768
 input_session_request
 channel 0: new [server-session]
 session_new: allocate (allocated 0 max 10)
 session_unused: session id 0 unused
 session_new: session 0
 session_open: channel 0
 session_open: session 0: link with channel 0
 server_input_channel_open: confirm session
 send packet: type 91
 receive packet: type 80
 server_input_global_request: rtype no-more-sessions@openssh.com want_reply 0
 receive packet: type 98
 server_input_channel_req: channel 0 request subsystem reply 1
 session_by_channel: session 0 channel 0
 session_input_channel_req: session 0 req subsystem
 subsystem request for sftp by user Administrator
 subsystem: exec() C:\\Windows\\System32\\OpenSSH\\sftp-server.exe
g session: subsystem 'sftp' for Administrator from 12.122.3.3 port 52369 id 0
 fd 9 setting O_NONBLOCK
 fd 10 setting O_NONBLOCK
 fd 11 setting O_NONBLOCK
 fd 12 setting O_NONBLOCK
 fd 13 setting O_NONBLOCK
 fd 14 setting O_NONBLOCK
 Executing command: "c:\\windows\\system32\\cmd.exe" /c "C:\\Windows\\System32\\OpenSSH\\sftp-server.exe" with 
 fd 4 setting TCP_NODELAY
 fd 11 is O_NONBLOCK
 fd 10 is O_NONBLOCK
 fd 13 is O_NONBLOCK
 send packet: type 99
```

**Comment by SilmorSenedlen** [comment]
I have the same problem described here:
#237

**Comment by voxsoftware** [comment]
same error

**Comment by dslater38** [comment]
From what I can tell, this seems to be a permissions issue. If I run:
sshfs-win.exe svc \sshfs.k\unixuser@ip_addr\. x: DOMAIN\user
then it works. However, 
net use x: \sshfs.k\unixuser@ip_addr\ gives a system error 67, as does using explorer to map a drive.

However, when I run net use x: \sshfs\unixuser@ip_addr\, I am prompted for a password, and I connect successfully. Then, looking at taskmgr, I discovered that when using net use, or explorer to map a drive, sshfs-win.exe, sshfs.exe, and ssh.exe are all running at the SYSTEM user. 

So, then I ran psexec -i -s cmd.exe to get a command shell running under the SYSTEM account. In that command shell, I tried running, from the SSHFS-WIn directory:
 .\ssh.exe unixuser@ip_addr 

And I saw this error:
Could not create directory '/home/SYSTEM/.ssh' (No such file or directory). I also tried running ssh.exe -i path_to_id_rsa_file, but always got a permission denied error trying to read the keyfile. It looks like the SYSTEM account has very limited access to the filesystem.

**Comment by calanor** [comment]
I have the same error.  I use auth with prompt password,  sshfs instead sshfs.k

In windows, ask for login and password and saw "Network error ....".  With "net use \sshfs\host X:" return error 67.  And using "C:\Program Files\SSHFS-Win\bin\sshfs-win.exe" svc \sshfs\user@host x:" i have a silent total, not prompt for password and if I type a password does nothing.

Running ssh.exe from C:\Program Files\SSHFS-Win\bin login without problem to my host.  I testted SFTP-Drive (free version) and also run perfectly.

Using:  
winsfp 1.8.20304 
sshfs 3.5.20357   
Windows 10 Enterprise 1909 (18363.1256).   

In other machine with windows 10 work perfectly.  What things could you look to see why it doesn't work?

**Comment by kaplan2539** [comment]
First of all, thanks Bill - this is great Software! Here's what I found out:

These two work without problem
```
net use X: \\sshfs\root@myhost!2233
net use X: \\sshfs.r\root@myhost!2233
```
NOTE: Taskmanager reveals `ssh.exe`, `sshfs.exe` and `sshfs-win.exe` all run as `SYSTEM` user.

Trying to use a public key auth fails:
```
net use X: \\sshfs.k\root@myhost!2233
net use X: \\sshfs.kr\root@myhost!2233
```
with this error message:
```
System error 67 has occurred.

The network name cannot be found.
```
Looking into the ssh server's log reveals the client is actually interrupting the connection.

OK, trying `sshfs-win svc \sshfs.kr\root@myhost!2233 X:` which also fails, but for a different reason:
On my Linux server running `/usr/sbin/sshd -ddd` says: `debug3: mm_answer_keyallowed: key 0xdeadcafec001 is not allowed`.

So, trying to get some information from the client side and run
```
sshfs-win svc \sshfs.kr\root@myhost!2233 X: %USERNAME% -d
```
which reveals:
```
SSHFS version 3.5.2
executing <ssh> <-x> <-a> <-oClearAllForwardings=yes> <-oUserKnownHostsFile=/dev/null> <-oStrictHostKeyChecking=no> <-oPreferredAuthentications=publickey> <-oIdentityFile="//moesbar/home$/k2539/.ssh/id_rsa"> <-oPort=2233> <-2> <root@myhost> <-s> <sftp>
Warning: Permanently added '[myhost]:2233' (ECDSA) to the list of known hosts.
root@myhost: Permission denied (publickey,password).
read: Connection reset by peer
```
Aha, `sshfs-win` is using `//moesbar/home$/k2539/.ssh/id_rsa` as private key and not `%USERPROFILE%\.ssh\id_rsa` as indicated in the documentation. On my system `%USERPROFILE%` is `C:\Users\k2539`. Don't ask me why I also have an ssh key sitting on some network share - while I'm having full admin rights the computer is maintained by the companies IT department and there seems to be a local and a network user profile. You might be running into different problems if you don't have an ssh key there or no network profile. What I'm trying to say is that running

```
sshfs-win svc \sshfs.kr\%SSH_USER%@%SSH_SERVER%!%SSH_PORT% X: %USERNAME% -d
```
might reveal some useful information.

And this is what actually allowed me to connect using public key auth after studying the `sshfs-win` source code:
```
sshfs-win svc \sshfs.kr\root@myhost!2233 X: USER_DONT_EXIST -d -oIdentityFile=C:/Users/k2539/.ssh/id_rsa
```

Some explanation:
You have to pass in a user name that doesn't exist on your system (e.g. `USER_DONT_EXIST`), because then `sshfs-win` does not pass the `oIdentityFile=...` option on to `sshfs`. You're responsible for that now, that's why you need to pass `-oIdentityFile=C:/where/ever/your/key/is/id_rsa`.

NOTE: after sucessfully connecting, a look at the Task Manager shows me that `ssh.exe`, `sshfs.exe` and `sshfs-win.exe` all run as %USERNAME%. I have the following suspicion: the `net use X: \\sshfs.k\` calls fail, because they are searching for the id_rsa key some where in the `SYSTEM` users profile.

Anyway, I hope this is useful for others and am looking forward to the next release of sshfs Bill!

**Comment by kaplan2539** [comment]
I'd like to share this batch script that is based on Winsfp's `fsreg.bat` which works with the latest sshfs-win version (as of today). It allows to register a specific ssh key as a filesystem in Winsfp. Very useful if you want to use several ssh keys in non standard locations. It should also work with ssh keys other than rsa, eg. id_ed25519. It only supports absolute paths on the server as of now, switching to relative paths is changing one number in the code below...

TL;DR: Very nasty hack. Save the batch file below as `reg_key.bat` to your hard drive. It might help you solve some of the problems with sshfs using public key authentication.

Usage examples:
```
reg_key id_rsa %USERPROFILE%\.ssh\id_rsa
reg_key id_rsa2 %USERPROFILE%\.ssh\id_rsa2
reg_key id_ed25519 D:\master_key\id_ed25519
```

Now mount a drive:
```
net use X: \\id_rsa\USER@HOST!PORT\ABSOLUTE_PATH
```
or
```
net use Y: \\id_ed25519@OTHER_SERVER!PORT\ABSOLUTE_PATH
```

CODE:
```
@echo off

setlocal
setlocal EnableDelayedExpansion

set RegKey=HKLM\Software\WinFsp\Services

if not X%1==X-u (
	set unreg=0

	if not X%1==X set fsname=%1
	set fsexec="C:\Program Files\SSHFS-Win\bin\sshfs-win.exe"
	set KEY_PATH=%~fp3
	set fscmdl="svc %%1 %%2 USER_DONT_EXIST -oIdentityFile=%KEY_PATH:\=/%"
	set fssecu="D:P(A;;RPWPLC;;;WD)"

	if X%2==X goto usage
	if X!fsname!==X goto usage
	if not exist !fsexec! (echo "ERROR: %fsexec% not found" & exit /b 2 )
	if not exist %KEY_PATH% (echo "ERROR: %KEY_PATH% not found" & exit /b 2 )

	reg add !RegKey!\!fsname! /v Executable /t REG_SZ /d !fsexec! /f /reg:32
	reg add !RegKey!\!fsname! /v CommandLine /t REG_SZ /d !fscmdl! /f /reg:32
	reg add !RegKey!\!fsname! /v JobControl /t REG_DWORD /d 1 /f /reg:32
	reg add !RegKey!\!fsname! /v Credentials /t REG_DWORD /d 0 /f /reg:32
	reg add !RegKey!\!fsname! /v sshfs.rootdir /t REG_DWORD /d 1 /f /reg:32
	if not X!fssecu!==X reg add !RegKey!\!fsname! /v Security /t REG_SZ /d !fssecu! /f /reg:32
	
	NET STOP WinFsp.Launcher
	NET START WinFsp.Launcher
) else (
	set unreg=1

	if not X%2==X set fsname=%2

	if X!fsname!==X goto usage

	reg delete !RegKey!\!fsname! /f /reg:32
)

exit /b 0

:notfound
echo File !fsexec! not found >&2
exit /b 2

:usage
echo usage: %~n0 NAME SSH_IDENTITY_FILE >&2
echo usage: %~n0 -u NAME >&2
exit /b 2
```

**Comment by billziss-gh** [comment]
@kaplan2539 great tip. FYI, you do not need to stop and restart the Launcher for the registry changes to be seen.

**Comment by ForbiddenEra** [comment]
I may be running into this issue as well..
`
Edition	Windows 10 Pro
Version	Dev
Installed on	‎2021-‎05-‎14
OS build	21376.1
Experience	Windows 10 Feature Experience Pack 321.11803.0.3
`

`
C:\Program Files\SSHFS-Win\bin>net use H: \\sshfs.r\kiosk@10.0.0.164\www
The password is invalid for \\sshfs.r\kiosk@10.0.0.164\www.

Enter the user name for 'sshfs.r': kiosk
Enter the password for sshfs.r:
System error 67 has occurred.

The network name cannot be found.
`

`
C:\Program Files\SSHFS-Win\bin>sshfs-win svc \sshfs.r\kiosk@10.0.0.164 H:
`

appears to hang forever.. also tried giving it my local username as per the `-h`, same result.

`
C:\Program Files (x86)\WinFsp\bin>diag
WINFSP INSTALLATION DIRECTORY AND LAUNCHER REGISTRATIONS

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp
    InstallDir    REG_SZ    C:\Program Files (x86)\WinFsp\

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x1

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.k
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x0

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.kr
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x0
    sshfs.rootdir    REG_DWORD    0x1

HKEY_LOCAL_MACHINE\SOFTWARE\WinFsp\Services\sshfs.r
    Executable    REG_SZ    C:\Program Files\SSHFS-Win\bin\sshfs-win.exe
    CommandLine    REG_SZ    svc %1 %2 %U
    Security    REG_SZ    D:P(A;;RPWPLC;;;WD)
    JobControl    REG_DWORD    0x1
    Credentials    REG_DWORD    0x1
    sshfs.rootdir    REG_DWORD    0x1


WINFSP DLL REGISTRATIONS

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order
    ProviderOrder    REG_SZ    P9NP,WinFsp.Np,RDPNP,LanmanWorkstation,webclient


HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\WinFsp.Np\NetworkProvider
    Name    REG_SZ    Windows File System Proxy
    ProviderPath    REG_SZ    C:\Program Files (x86)\WinFsp\bin\winfsp-x64.dll
    DeviceName    REG_SZ    \Device\WinFsp.Mup


HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\Application\WinFsp
    EventMessageFile    REG_SZ    C:\Program Files (x86)\WinFsp\bin\winfsp-x64.dll
    TypesSupported    REG_DWORD    0x7


WINFSP FSD CONFIGURATION AND STATUS

SERVICE_NAME: WinFsp
        TYPE               : 2  FILE_SYSTEM_DRIVER
        STATE              : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, IGNORES_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
[SC] QueryServiceConfig SUCCESS

SERVICE_NAME: WinFsp
        TYPE               : 2  FILE_SYSTEM_DRIVER
        START_TYPE         : 3   DEMAND_START
        ERROR_CONTROL      : 1   NORMAL
        BINARY_PATH_NAME   : \??\C:\Program Files (x86)\WinFsp\bin\winfsp-x64.sys
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : WinFsp
        DEPENDENCIES       :
        SERVICE_START_NAME :

D:(A;;LCRP;;;WD)(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)

WINFSP LAUNCHER SERVICE CONFIGURATION AND STATUS

SERVICE_NAME: WinFsp.Launcher
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 1067  (0x42b)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
[SC] QueryServiceConfig SUCCESS

SERVICE_NAME: WinFsp.Launcher
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 0   IGNORE
        BINARY_PATH_NAME   : "C:\Program Files (x86)\WinFsp\bin\launcher-x64.exe"
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : WinFsp.Launcher
        DEPENDENCIES       :
        SERVICE_START_NAME : LocalSystem

D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)

OS INFORMATION

Host Name:                 FORBIDDENERA
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.21376 N/A Build 21376
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Forbidden Era
Registered Organization:
Product ID:                00330-80000-00000-AA846
Original Install Date:     2021-05-14, 9:48:54 AM
System Boot Time:          2021-05-14, 9:45:01 AM
System Manufacturer:       Acer
System Model:              Predator PO3-600
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 158 Stepping 13 GenuineIntel ~3000 Mhz
BIOS Version:              American Megatrends Inc. R01-C1, 2020-04-08
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume5
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-07:00) Mountain Time (US & Canada)
Total Physical Memory:     12,204 MB
Available Physical Memory: 1,455 MB
Virtual Memory: Max Size:  36,780 MB
Virtual Memory: Available: 15,788 MB
Virtual Memory: In Use:    20,992 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\FORBIDDENERA
Hotfix(s):                 1 Hotfix(s) Installed.
                           [01]: KB5003252
Network Card(s):           6 NIC(s) Installed.
                           [01]: Wintun Userspace Tunnel
                                 Connection Name: OpenVPN Wintun
                                 Status:          Media disconnected
                           [02]: TAP-Windows Adapter V9
                                 Connection Name: Local Area Connection 2
                                 Status:          Media disconnected
                           [03]: Intel(R) Wireless-AC 9560 160MHz
                                 Connection Name: Wiffy
                                 Status:          Media disconnected
                           [04]: TAP-Windows Adapter V9 for OpenVPN Connect
                                 Connection Name: Local Area Connection
                                 Status:          Media disconnected
                           [05]: Realtek Gaming GbE Family Controller
                                 Connection Name: LAN
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 10.0.0.10
                           [06]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection 2
                                 Status:          Media disconnected
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
`

`
C:\Program Files (x86)\WinFsp\bin>launchctl-x64.exe start \sshfs.r\kiosk@10.0.0.164 H:
KO CallNamedPipe = 2
`

Did windows update break us?

**Comment by mElBalkini** [comment]
I had the same issue as @kaplan2539 
ssh was trying to get `.ssh` folder from my corporate network drive. However, I didn't want to change the registry.

Here is how I fixed it:

Go to `C:\Program Files\SSHFS-Win\etc` and create a `ssh_config` file with the following content
```
Host *
  IdentityFile C:\Users\%u\.ssh\id_rsa
  userknownhostsfile C:\Users\%u\.ssh\known_hosts
```
Note that `ssh.exe` converts `%u` to the current username.
Thanks a lot @kaplan2539  for narrowing this issue down

**Comment by EsherionM** [comment]
> From what I can tell, this seems to be a permissions issue. If I run: sshfs-win.exe svc \sshfs.k\unixuser@ip_addr. x: DOMAIN\user then it works. However, net use x: \sshfs.k\unixuser@ip_addr\ gives a system error 67, as does using explorer to map a drive.
> 
> However, when I run net use x: \sshfs\unixuser@ip_addr, I am prompted for a password, and I connect successfully. Then, looking at taskmgr, I discovered that when using net use, or explorer to map a drive, sshfs-win.exe, sshfs.exe, and ssh.exe are all running at the SYSTEM user.
> 
> So, then I ran psexec -i -s cmd.exe to get a command shell running under the SYSTEM account. In that command shell, I tried running, from the SSHFS-WIn directory: .\ssh.exe unixuser@ip_addr
> 
> And I saw this error: Could not create directory '/home/SYSTEM/.ssh' (No such file or directory). I also tried running ssh.exe -i path_to_id_rsa_file, but always got a permission denied error trying to read the keyfile. It looks like the SYSTEM account has very limited access to the filesystem.

I have created /home/SYSTEM/.ssh with my admin user and WOW all works. Thanks for telling the Path

**Comment by perrocontodo** [comment]
Thanks @mElBalkini ! That did the trick. For sake of documentation, I'm connecting to a Vagrant box in my machine. By creating a file in C:\Program Files\SSHFS-Win\etc\ssh_config with the following content, I was able to map the drive using explorer.exe (replace <username> with the appropriate user):

```
Host localhost_vagrant
  HostName localhost
  IdentityFile C:\Users\<username>\.vagrant.d\insecure_private_key
  userknownhostsfile C:\Users\<username>\.ssh\known_hosts
  User vagrant
  Port 2222
```

**Comment by dflock** [comment]
Well, that's great if you're local admin and allowed to write to the `C:\Program Files\SSHFS-Win\etc` folder :(

**Comment by ForbiddenEra** [comment]
> I had the same issue as @kaplan2539 ssh was trying to get `.ssh` folder from my corporate network drive. However, I didn't want to change the registry.
> 
> Here is how I fixed it:
> 
> Go to `C:\Program Files\SSHFS-Win\etc` and create a `ssh_config` file with the following content
> 
> ```
> Host *
>   IdentityFile C:\Users\PUT_YOUR_USERNAME_HERE\.ssh\id_rsa
>   userknownhostsfile C:\Users\PUT_YOUR_USERNAME_HERE\.ssh\known_hosts
> ```
> 
> Replace `PUT_YOUR_USERNAME_HERE`. `ssh.exe` converting `%u` to the current username doesn't work! Thanks a lot @kaplan2539 for narrowing this issue down

might have to try this; kind of annoying to have a bunch of `.bat` files to mount w/open terminal windows while they're in use!

**Comment by ForbiddenEra** [comment]
> Well, that's great if you're local admin and allowed to write to the `C:\Program Files\SSHFS-Win\etc` folder :(

How did you install SSHFS then..?

If it's installed by your company/IT or something, then perhaps you should just point an admin to the fix since there's no sense in having it installed if it doesn't even work!

Otherwise, if you were able to install, then you should be able to make changes to that folder.

**Comment by dflock** [comment]
> How did you install SSHFS then..?

I used to be local admin on this box, but things got more locked down by IT, and now I'm not.

**Comment by johnburnett** [comment]
If it helps anyone, I ran into the "error 67 The network name cannot be found" issue as well when trying to mount via "`net use`" in a cmd prompt with the `sshfs.k` auth type.  However, directly calling `sshfs-win` manually worked fine.

The kicker is that using procmon to look at the "`net use`" launch of the `sshfs-win` child process showed that the resulting command line was exactly the same as the working manual `sshfs-win` case.  The difference between the two cases is that the working case was running under my user account, and the broken case was running under the SYSTEM account due to the "`net use`" command.

The root cause turned out to be that my user account's `%USERPROFILE%\.ssh\` directory is a symlink to a separate network location that's mounted only for my account (it's actually a Google Drive virtual drive).  The SYSTEM account doesn't have that network location mounted, and I think *that* was what the "network name cannot be found" error above was referring to (it was still finding my account's `.ssh` dir, but it was a symlink pointing to an unavailable network resource.)  For me, the fix would be to make my `.ssh` dir available locally in some manner (by either making the `.ssh` dir a real dir not a symlink, or make a copy of it somewhere locally that could be pointed at via ssh_config/registry settings that others have suggested above).

**Comment by JFLarvoire** [comment]
I had the same problem, but with a twist:

I have two Raspberry Pi servers configured similarly with Raspbian.
One (called pi) is at home; The other (called pi3) is in my parent's home.
Connecting sshfs-win to Pi on my home network succeeds.
Connecting sshfs-win to Pi3 on the remote network failed with error 67, and `net use R: \\sshfs.r\...` exited with errorlevel 2.
BUT, despite the error message and non-0 exit code, the connection actually worked! The R: drive was accessible normally. 😲
This happened the same way on two distinct clients, one running Windows 10, and the other Windows 11.

I first tried @ForbiddenEra's suggestion to create a "C:\Program Files\SSHFS-Win\etc\ssh_config" file with my own user name, but this has no effect on the issue.
Restarting WinFsp.Launcher does not help.

Like @calanor, running the "C:\Program Files\SSHFS-Win\bin\sshfs-win.exe" command hangs with total silence.
But, contrary to him, typing the password at this stage unblocks the program, and the connection succeeds.
Comparing the debug output of sshfs-win.exe for the two servers showed no difference.

I did not try the changes proposed by @kaplan2539, because I'm using passwords, not ssh keys.

Then I began to suspect some kind of ssh handshake validation issue:
In the succeeding case with Pi, the connection was very quick.
In the (not quite) failing case with Pi3, the connection took much longer.
Initially I had thought this discrepancy was due to a slow Internet connection.
Anyway I then tried measuring the duration of that connection:
While running `net use R: \\sshfs.r\...` in a command window, I ran `net use` repeatedly in another.
And I saw that the first command timed out with error 67 after 20 seconds, whereas the R: drive appeared only after 22 seconds in the second!
So the error 67 was real: The network drive was not mounted when the first command exited. It only got mounted 2 seconds later.
Which is short enough that I had not noticed the delay before.

Searching in `net help use`, I found option `/skipcertcheck` which looked promising.
It proved a bit difficult to use, because it requires also using `/transport:quic`, and passing an IP address instead of a DNS name.
But finally, running `net use R: \\sshfs.r\NAME@IPADDR\PATHNAME /user:NAME /transport:quic /skipcertcheck` worked great:
The connection to Pi3 was immediate, and with no error. 😊

Later, I found that `net use R: \\sshfs.r\NAME@IPADDR\PATHNAME /user:NAME` worked just as well! What?
Double-checking with sshfs-win.exe, I saw that the behaviour was the same:
Using the IP address, the connection to Pi3 was fast.
Using the DNS name, it took more than 20 seconds.
So `/skipcertcheck` was useless, it's using the IP address that was the critical factor!

So, for me, the root cause of the error 67 problem does not come from sshfs-win, nor from slow access over the Internet.
It comes from the absence of a certificate for the domain name I'm using to connect remotely to Pi3.
I'm not sure if this is something I failed to configure on Pi3, or if this is the dynamic DNS provider that failed to do something on its side.

Anyway, sshfs-win.exe does have its own share of issues:

 * It would be nice if sshfs-win.exe displayed a detailed message in debug mode when such a timeout occurs.
   That would have saved me quite a lot of time while investigating the issue.
 * sshfs-win.exe debug output says it's running ssh with option `-oStrictHostKeyChecking=no`.
   I would have thought this would have avoided the issue, but it does not.
 * `sshfs-win.exe --help` mentions `SSHFS_OPTIONS       additional options to pass to SSHFS`
   The help message should list them all, or at least explain where are they documented!
 * Is there one such option that would prevent the timeout when using the DNS name?
   If so which one?
   Else please consider adding one!
   (I really need the Dynamic DNS name, because my parents' home IP address is not guarantied to be fixed!)
 * It'd be nice if there were a way to pass share-specific mount options.
   Possibly via a config file in `%HOME%\.config\sshfs\sshfs\`?

**Comment by ForbiddenEra** [comment]
> I first tried @ForbiddenEra's suggestion 

That wasn't my suggestion, it was @mElBalkini's; I was just quoting it from someone else - I never tried it really.

My solution is just having batch files to mount and leaving the console windows for those mounts open as needed. PITA but works.

**Comment by joaomamede** [comment]
I have the same problem. 
It used to work. I had to change my server location, used another Macaddress and changed the IP in the network. Now I can't use net use \\sshfs.r  while \\sshfs works.

Through the explorer it works right now. It wasn't working for a while.

**Comment by emv33** [comment]
Dozens of workarounds proposed, none that really works :(

Software seems to be abandoned, too, with no releases in four years. Anybody knows of any good alternative? I tried Mountain Duck and it was nothing but unreliable.

**Comment by mbertol4** [comment]
I used RaiDrive (https://www.raidrive.com/download) as a commercial tool to handle this.

**Comment by dflock** [comment]
https://rclone.org/commands/rclone_mount/

**Comment by aserrallerios** [comment]
Based on my requirements:
* be able to use a key file protected with passphrase
* be able to execute programs from the drive as administrator
* be able to see the mount in my regular session user

I'm using the following discussed solution:

```
& Y:\SysinternalsSuite\PsExec64.exe -s -d -i 'C:\Program Files\SSHFS-Win\bin\sshfs-win.exe' svc \sshfs.kr\root@192.168.1.5!22/some/path X: USER_DONT_EXIST -d -oIdentityFile="C:/mykey.openssh.key"
```

~~This has a clear drawback: the mount is owned by SYSTEM, so we cannot modify or create any files, regardless of the remote user permissions.~~ This was a wrong assumption. The remote permissions are preserved, the only limitation is that there's no user mapping. There's only SYSTEM and "everyone else", and ALL the files/folders are owned by SYSTEM, so your ssh user will always belong to "everyone else", even when in the remote server the ssh user was the owner of some folders/files.

It also has another drawback, which is that "Run as Administrator" doesn't seem to work seamlessly, as I still receive the typical error of "cannot access":
![image](https://github.com/user-attachments/assets/fd9a2ba9-170e-4e78-8a92-44948ab13729)
But at least I can run programs as administrator from the Admin Console. The problem seems to be that the mount is not using the drive letter (X: in this case) but for some reason it's using the remote full address instead.

Finally, probably because of the usage of `PsExec64`, I need to type my passphrase for every connection, even though I have the OpenSSH up and running, and the key has been loaded:

```
Admin console > ssh-add -l
4096 SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx @mykey.openssh.key (RSA)
```

I'd appreciate if anyone found and could share a more ergonomic solution.

**Comment by aserrallerios** [comment]
Something interesting and weird at the same time about "Run as Administrator" is that if the mount command has a PATH I got the "Windows cannot access" error, but if I use a path-less command, it works perfectly (i.e. I can "Run as Administrator" from the "File Explorer" and I get no error):

```
& Y:\SysinternalsSuite\PsExec64.exe -s -d -i 'C:\Program Files\SSHFS-Win\bin\sshfs-win.exe' svc \sshfs.kr\root@192.168.1.5!22 X: USER_DONT_EXIST -d -oIdentityFile="C:/mykey.openssh.key"
```

**Comment by nghiatr84** [comment]
In [the main README.md file](https://github.com/winfsp/sshfs-win?tab=readme-ov-file#unc-syntax) it said 

> To specify a different specific key, define an alias of the HOST with the specific private ssh key you want to use in the ssh config.

This mislead me to think I should configure ssh_config in the ~/.ssh folder, not the C:/Program Files/SSHFS-Win/etc folder... I think it should be clarified where should the ssh config go.

**Comment by vico93** [comment]
I tried to follow @kaplan2539 's trick [here](https://github.com/winfsp/sshfs-win/issues/155#issuecomment-777466963) but the .bat seems to get an error at the end of the reg_key bat... was giving up until i found (in another issue) the blessed @perrocontodo [comment](https://github.com/winfsp/sshfs-win/issues/155#issuecomment-1316261288) just below. My god, i should roll more before attempting a fix 🤦🏻

> In [the main README.md file](https://github.com/winfsp/sshfs-win?tab=readme-ov-file#unc-syntax) it said
> 
> > To specify a different specific key, define an alias of the HOST with the specific private ssh key you want to use in the ssh config.
> 
> This mislead me to think I should configure ssh_config in the ~/.ssh folder, not the C:/Program Files/SSHFS-Win/etc folder... I think it should be clarified where should the ssh config go.

@nghiatr84 totally agreed with you mate! If this matter was better explained on README many like me would not bang their heads or use half-baked solutions like SSHFS-Win Manager that have lots of UI issues and don't mount as network places.

**Comment by samuliasmala** [comment]
I ran into the same issue with `sshfs.k` when using key-only authentication and `id_ed25519` key name.  `ssh samuli@<myhost>` and `ssh dev` worked but both of these

```text
net use X: \\sshfs.k\samuli@<myhost>
net use X: \\sshfs.k\dev
```

failed with

```
System error 67 has occurred.

The network name cannot be found.
```

My `%USERPROFILE%\.ssh\config` was

```sshconfig
Host dev
  HostName <myhost>
  User samuli
  IdentityFile C:/Users/samul/.ssh/id_ed25519
  IdentitiesOnly yes
```

**Solution:**  
The cause was that Windows OpenSSH reads `%USERPROFILE%\.ssh\config` whereas SSHFS-Win was reading `C:\Program Files\SSHFS-Win\etc\ssh_config`. After copying my SSH configuration to the SSHFS-Win config file, the mount started working:

```powershell
Copy-Item "$HOME\.ssh\config" "C:\Program Files\SSHFS-Win\etc\ssh_config"
```

This must be run from an elevated PowerShell prompt because the destination is under `Program Files`.

After that, this worked:

```powershell
net use X: \\sshfs.k\dev
The command completed successfully.
```

So if `ssh <alias>` works, but `\\sshfs.k\<alias>` returns error 67, check which SSH config file SSHFS-Win is actually using.
